### ISSUE TITLE
Vulkan: Vertex colors glitch on 3D model due to automatic mesh LOD

### ISSUE THREAD

**Issue by kevinloustau** [godotengine/godot#57608]
___

***Bugsquad note:** This issue has been confirmed several times already. No need to confirm it further.*

___

### Godot version

v4.0.alpha.custom_build [2aee84c75]

### System information

Windows 11

### Issue description

https://user-images.githubusercontent.com/4235617/152444746-6d3e3279-3056-4ccc-a528-b92741715a07.mp4

Using vertex painting on a model. If I zoom/de-zoom on my model, there is a glitch as you can see above. 
It is working well in 3.4.2.

(Is it a LOD issue?)

### Steps to reproduce

- create a 3d mesh model, then add paint vertex. (or use the one attatched)
- import the model in **godot** (master). The previewer is working well.
- import the model in a **scene**, then zoom and dezoom on the model.

Files:
[cube vertex paint .blend + gltf.zip](https://github.com/godotengine/godot/files/7998926/cube.vertex.paint.blend.%2B.gltf.zip)
 

### Minimal reproduction project

[vertex painting 3.4.2.zip](https://github.com/godotengine/godot/files/7998904/vertex.painting.3.4.2.zip)
[vertex painting 4.0.zip](https://github.com/godotengine/godot/files/7998905/vertex.painting.4.0.zip)

**Comment by Calinou** [comment]
> (Is it a LOD issue?)

Yes, it's very likely. The automatic mesh LOD does support vertex colors, but it's possible that your particular mesh isn't well handled.

As a workaround, set **Threshold Pixels** to `0.0` in the Project Settings to disable the rendering of automatic mesh LOD.

**Comment by fire** [comment]
If there are not enough vertices for the vertex color to look correct, do we interpolate?, do we disable lod?, do we do some magical mesh processing?

**Comment by kevinloustau** [comment]
From a user point of view, do this use case (low poly + vertex color) is a niche? 
If yes, to disable LOD in settings seems ok. As @fire said, to post a documentation note seems enough.

**Comment by Calinou** [comment]
> From a user point of view, do this use case (low poly + vertex color) is a niche?

When working with low-poly meshes, automatic mesh LOD won't benefit you much. It makes sense to disable LOD for all your imported meshes in the **Import Defaults** tab of the Project Settings.

> If there are any non-white data in the vertex color channel do we disable lod? This doesn't feel like the right design.

LOD on vertex-colored meshes should definitely be kept in. I wonder if meshoptimizer could be modified to take vertex colors into account

**Comment by fire** [comment]
I think you can assign the 4 rgba channels to the attribute simplifier. Unknown worth.

**Comment by fire** [comment]
I think we can resolve this with a documentation entry.

**Comment by lyuma** [comment]
Can we investigate if it's possible to quickly mitigate this problem by passing vertex colors to meshopt?

**Comment by RadiantUwU** [comment]
Any updates? This issue has not been looked at in over 2 years.

**Comment by fire** [comment]
We've upgraded the LOD tooling in meshoptimizer but haven't tested the original cases.

**Comment by aristides86** [comment]
In v4.3.stable.official [77dcf97d8] it just removes vertex colors alltogether. The temp workaround is to disable LOD generation on import.

**Comment by catprisbrey** [comment]
Godot 4.4RC1, Blender 4.3, UbuntuStudio 24.04.

I want to add to this since I've been experiencing it a lot lately. I'll share a GIF since mine is about as minimal of an example of it as i can imagine. I have a 64x64m plane in blender, vertec colors painted in a square-ish RGB pattern. With LOD generation disabled it works fine, enabled it heavily distorts the color data. 

![Image](https://github.com/user-attachments/assets/41150500-abb2-457c-a781-7558c9f1c7dd)

**Comment by Megalukes** [comment]
@catprisbrey, same thing here. Using Godot 4.4.1.

**Comment by zeux** [comment]
I think this can be closed as part of 4.5 (most of these are fixed by https://github.com/godotengine/godot/pull/106446, if any lingering issues remain it'll be better to create a new issue with more focused examples)

**Comment by Calinou** [comment]
Closing per the above comment.
